File: android/src/main/java/lt/miror/kustomer/RNKustomerModule.java
package lt.miror.kustomer;

import com.kustomer.kustomersdk.Kustomer;

import android.app.Activity;

import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;


public class RNKustomerModule extends ReactContextBaseJavaModule {

  public KustomerModule(ReactApplicationContext reactContext) {
    super(reactContext);
  }

  @Override
  public String getName() {
    return "KustomerModule";
  }

  @ReactMethod
  public void presentSupport() {
    final Activity activity = getCurrentActivity();
    Kustomer.showSupport(activity);
  }

  @ReactMethod
  public void presentKnowledgeBase() {
    final Activity activity = getCurrentActivity();
    Kustomer.presentKnowledgeBase(activity);
  }

  @ReactMethod
  public void presentCustomWebPage(String url) {
    final Activity activity = getCurrentActivity();
    Kustomer.presentCustomWebPage(activity, url);
  }

  @ReactMethod
  public void resetTracking() {
    Kustomer.resetTracking();
  }

  @ReactMethod
  public void identify(String hash) {
    Kustomer.identify(hash);
  }

  @ReactMethod
  public void setCurrentPageName(String pageName) {
    Kustomer.setCurrentPageName(pageName);
  }

}

File: ios/RNKustomerModule/RNKustomerModule.m
//
//  RNKustomerModule.m
//  RNKustomerModule
//
//  Created by Aleksandras Smirnovas on 22/07/2018.
//  Copyright © 2018 Miror MB. All rights reserved.
//
#import "RNKustomerModule.h"
#import <Kustomer/Kustomer.h>
@implementation RNKustomerModule
RCT_EXPORT_METHOD(presentSupport)
{
  dispatch_async(dispatch_get_main_queue(), ^{
    [Kustomer presentSupport];
  });
}
RCT_EXPORT_METHOD(presentKnowledgeBase)
{
  dispatch_async(dispatch_get_main_queue(), ^{
    [Kustomer presentKnowledgeBase];
  });
}
RCT_EXPORT_METHOD(presentCustomWebPage:(NSString *) url)
{
  dispatch_async(dispatch_get_main_queue(), ^{
    [Kustomer presentCustomWebPage:url];
  });
}
RCT_EXPORT_METHOD(resetTracking)
{
  dispatch_async(dispatch_get_main_queue(), ^{
    [Kustomer resetTracking];
  });
}
RCT_EXPORT_METHOD(identify:(NSString *) hash)
{
  dispatch_async(dispatch_get_main_queue(), ^{
    [Kustomer identify:hash];
  });
}
RCT_EXPORT_METHOD(setCurrentPageName:(NSString *) pageName)
{
  dispatch_async(dispatch_get_main_queue(), ^{
      [Kustomer setCurrentPageName:pageName];
  });
}
@end
